src/libhirte/log/log.h

```diff
@@ -0,0 +1,110 @@
+#pragma once
+
+#include <stdarg.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <systemd/sd-journal.h>
+
+#include "libhirte/common/common.h"
+
+typedef enum LogLevel {
+        LOG_LEVEL_DEBUG,
+        LOG_LEVEL_INFO,
+        LOG_LEVEL_WARN,
+        LOG_LEVEL_ERROR,
+} LogLevel;
+
+static const char *log_level_strings[] = { "DEBUG", "INFO", "WARN", "ERROR" };
+
+const char *log_level_to_string(LogLevel l) {
+        return log_level_strings[l];
+}
+
+
+typedef void (*LogFn)(LogLevel lvl, const char *msg, const char *data);
+
+void hirte_log_to_journald(LogLevel lvl, const char *msg, const char *data) {
+        // clang-format off
+        sd_journal_send(
+                "HIRTE_LOG_LEVEL=%s", log_level_to_string(lvl),
+                "HIRTE_MESSAGE=%s", msg,
+                "HIRTE_DATA=%s", data,
+                NULL);
+        // clang-format on
+}
+
+void hirte_log_to_stdout(LogLevel lvl, const char *msg, const char *data) {
+        time_t t = time(NULL);
+        char timebuf[9];
+        timebuf[strftime(timebuf, sizeof(timebuf), "%H:%M:%S", localtime(&t))] = '\0';
+
+        // clang-format off
+        printf("%s %s\t{msg: %s, data: %s}\n", 
```

Comment:
Should we maybe log to stderr instead? I'm not sure what is most useful.

Reply:
Good question. Maybe this should be dependent on the log level? For example, debug and info logs should go to stdout, but warnings and errors to stderr?

Reply:
Good point. I changed it accordingly.